src/openms/source/TRANSFORMATIONS/FEATUREFINDER/ElutionModelFitter.cpp

```diff
@@ -113,16 +116,97 @@ double ElutionModelFitter::calculateFitQuality_(const TraceFitter* fitter,
 }
 
 
+void ElutionModelFitter::fitAndValidateModel_(
+  TraceFitter* fitter, MassTraces& traces, Feature& feature,
+  double region_start, double region_end, bool asymmetric,
+  double area_limit, double check_boundaries)
+{
+  bool fit_success = true;
+  try
+  {
+    fitter->fit(traces);
+  }
+  catch (Exception::UnableToFit& except)
+  {
+    OPENMS_LOG_ERROR << "Error fitting model to feature '"
+                     << feature.getUniqueId() << "': " << except.getName()
+                     << " - " << except.getMessage() << endl;
+    fit_success = false;
+  }
+
+  // record model parameters:
+  double center = fitter->getCenter(), height = fitter->getHeight();
+  feature.setMetaValue("model_height", height);
+  feature.setMetaValue("model_FWHM", fitter->getFWHM());
+  feature.setMetaValue("model_center", center);
+  feature.setMetaValue("model_lower", fitter->getLowerRTBound());
+  feature.setMetaValue("model_upper", fitter->getUpperRTBound());
+  if (asymmetric)
+  {
+    EGHTraceFitter* egh = static_cast<EGHTraceFitter*>(fitter);
```

Comment:
these casts smell bad but not sure if its worth to refactor the TraceFitter class

Reply:
That's how it's been before, so maybe an issue for another time.